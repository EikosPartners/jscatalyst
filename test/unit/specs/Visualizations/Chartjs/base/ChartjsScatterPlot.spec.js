import Component from '@/components/Visualizations/Chartjs/base/ChartjsScatterChart'
import { shallow, mount } from '@vue/test-utils'
import { createRenderer } from 'vue-server-renderer'
import Vue from 'vue'
import ChartjsHelper from '@/common/chartjsHelper.js'


const data = [{"x": 4,"y": 10,"label": "value 1"},{"x": 8,"y": 5,"label": "value 2"},{"x": 10,"y": 10,"label": "value 3"}]
const props = new ChartjsHelper().scatterParser(data, ["#3e95cd", "#8e5ea2","#3cba9f","#e8c3b9","#c45850"]);


describe('Chartjs Scatter Plot', () => {
    let wrapper
    beforeEach(() => {
      wrapper = mount(Component, {
          propsData: {
              chartData: props
         },
         attachToDocument: true
      })
    })

    afterEach(() => {
        wrapper.destroy()
    })

    it('it renders at all', () => {
        expect(wrapper.contains('canvas')).toBe(true)
        expect(wrapper.html()).toContain('<canvas')
    })

    it('has the right prop data', () => {
      expect(wrapper.props()).toHaveProperty('chartData.datasets')
      expect(wrapper.props().chartData.datasets).toHaveLength(3)
      expect(wrapper.props().chartData.datasets[0].data).toContainEqual({"x": 4, "y": 10})
    })
})

describe('Chartjs Scatter Plot, snapshot', ()=>{
    let wrapper

    beforeEach(() => {
        wrapper = mount(Component, {
            propsData: {
                chartData: props
           },
           attachToDocument: true
        })
    })

    it('matches snapshot', () => {
        let renderer = createRenderer()
        return Vue.nextTick()
            .then(function(){
                renderer.renderToString(wrapper.vm, (err, str) => {
                  if (err) throw new Error(err)
                  expect(str).toMatchSnapshot()
                })
            })

    })

})
