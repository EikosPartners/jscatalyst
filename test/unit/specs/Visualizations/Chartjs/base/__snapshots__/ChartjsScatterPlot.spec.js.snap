// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chartjs Scatter Plot, snapshot matches snapshot 1`] = `
<div>
    <canvas id="scatter-chart" width="400" height="400"></canvas>
</div>
`;
